syntax = "proto3";
package pt.ulisboa.tecnico.nameserver.contract;

// The following option causes top-level messages, enums, and services to be defined at the package level, 
// rather than inside an outer class named after the .proto file.
option java_multiple_files = true;



message RegisterRequest {
  string service_name = 1;
  string qualifier = 2;
  string server_address = 3;
}

message EmptyResponse {}

message LookupRequest {
  string service_name = 1;
  string qualifier = 2;
}

message LookupResponse {
  repeated string servers = 1;
}

message DeleteRequest {
  string service_name = 1;
  string server_address = 2;
}

service NameServerService {
  rpc register(RegisterRequest) returns (EmptyResponse);
  rpc lookup(LookupRequest) returns (LookupResponse);
  rpc delete(DeleteRequest) returns (EmptyResponse);
}
